src/sql/workbench/services/notebook/common/notebookServiceImpl.ts

```diff
@@ -484,4 +514,83 @@ export class NotebookService extends Disposable implements INotebookService {
 			}
 		});
 	}
+
+	async isNotebookTrustCached(notebookUri: URI, isDirty: boolean): Promise<boolean> {
+		if (notebookUri.scheme === Schemas.untitled) {
+			return true;
+		}
+
+		let cacheInfo = this.trustedNotebooksMemento.trustedNotebooksCache[notebookUri.toString()];
+		if (!cacheInfo) {
+			// This notebook was never trusted
+			return false;
+		}
+		// This was trusted. If it's not dirty (e.g. if we're not working on our cached copy)
+		// then should verify it's not been modified on disk since that invalidates trust relationship
+		if (!isDirty) {
+			// Check mtime against mtime on disk
+			let actualMtime: number = await this.getModifiedTimeForFile(notebookUri);
+			if (actualMtime > cacheInfo.mtime) {
```

Comment:
Ah nice, this answers a question I previously had.

Reply:
What was the question?